packages/editor/src/components/post-actions/utility-components.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, Children, Fragment } from '@wordpress/element';
+import {
+	privateApis as componentsPrivateApis,
+	Modal,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const {
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuItemV2: DropdownMenuItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+	kebabCase,
+} = unlock( componentsPrivateApis );
+
+// All the functions on this file are copied as from the dataviews packages,
+// I think the editor packages should not be using the dataviews packages directly,
+// and the dataviews package should not be using the editor packages directly,
+// so duplicating the code here seems like the least bad option.
+
+// Copied as is from packages/dataviews/src/item-actions.js
+function DropdownMenuItemTrigger( { action, onClick } ) {
+	return (
+		<DropdownMenuItem
+			onClick={ onClick }
+			hideOnClick={ ! action.RenderModal }
+		>
+			<DropdownMenuItemLabel>{ action.label }</DropdownMenuItemLabel>
+		</DropdownMenuItem>
+	);
+}
+
+// Copied as is from packages/dataviews/src/item-actions.js
+function ActionWithModal( { action, item, ActionTrigger } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const actionTriggerProps = {
+		action,
+		onClick: () => setIsModalOpen( true ),
+	};
+	const { RenderModal, hideModalHeader } = action;
+	return (
+		<>
+			<ActionTrigger { ...actionTriggerProps } />
+			{ isModalOpen && (
+				<Modal
+					title={ action.modalHeader || action.label }
+					__experimentalHideHeader={ !! hideModalHeader }
+					onRequestClose={ () => {
+						setIsModalOpen( false );
+					} }
+					overlayClassName={ `editor-action-modal editor-action-modal__${ kebabCase(
+						action.id
+					) }` }
+				>
+					<RenderModal
+						items={ [ item ] }
+						closeModal={ () => setIsModalOpen( false ) }
+					/>
+				</Modal>
+			) }
+		</>
+	);
+}
+
+// Copied as is from packages/dataviews/src/utils.js
```

Comment:
Sometimes just copying + duplicating a small thing is nice, rather than defaulting to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I think it's underrated, even. But in this case I would either:

* Duplicate directly to where it will be used, i.e. `editor/src/components/post-actions/index.js` and don't export as a util.
* Make the decision to upgrade `WithDropDownMenuSeparators` to something reusable, in which case it belongs in `@wordpress/components` alongside `DropdownMenuSeparator`. Although, honestly, I'm more in favour of the first option for now.

Reply:
Hi @mcsf the reason for "post-actions/utility-components.js" was to have a file where everything inside it is duplicated code. But keeping it where it used also works and we can have a "start duplicated components" and an "end duplicated components" mark.